### PR TITLE
Improve comment and error message

### PR DESCRIPTION
**Pull request by basil**
Improve the comment from #246 to more accurately reflect what the code is doing, and improve the error message to state the name of the problematic entry when the condition is encountered.

CC @JLLeitschuh @daniel-beck

### CODE REVIEW

**Review by MarkEWaite — APPROVED** [review]

**Review by JLLeitschuh — APPROVED** [review]
Looks good to me!

**Comment by basil** [comment]
I'll wait a day or so to see if @daniel-beck has any feedback about the phrasing of the code comment prior to merging.

File: src/main/java/winstone/HostConfiguration.java [hunk 1]
```diff
@@ -281,9 +281,9 @@ protected File getWebRoot(File requestedWebroot, File warfile) throws IOExceptio
                     // If archive date is newer than unzipped file, overwrite
                     File outFile = new File(unzippedDir, elemName);
 
-                    // Fix Zip Slip vulnerability
+                    // Protect against Zip Slip attack
```
Comment (daniel-beck) [inline src/main/java/winstone/HostConfiguration.java:284]: This suggests there's a real attack possible when that's already slightly misleading.

```suggestion
                    // Disallow unzipping files outside the target dir)
```

**Review by daniel-beck — COMMENTED** [review]
No strong opinion either way.

FTR I've already rephrased https://github.com/jenkinsci/winstone/releases/tag/winstone-6.2 based on the conversation in the previous PR, as it's not a vulnerability for lack of impact: there is no increase of access, privileges gained, or other negative effect from the previous behavior.

**Review by basil — COMMENTED** [review]
Thanks for the feedback, Daniel!

**Pull request merged**